### PR TITLE
feat(examples): add `r/leon`

### PR DESCRIPTION
**Pull request by leohhhn**
## Description

This PR adds v1 of `r/leon` to the examples folder. It contains `r/leon/home` & `r/leon/config`.
 
For context - this is meant to be a personal realm (or set of realms) that every gno.land user should have at some point. Later, the idea is to have a [special user page](https://github.com/gnolang/gno/issues/2189) which will fetch the `Render()` of `r/username/home`. This will serve like the profile of the user; and the home realm will most likely be upgradeable down the line. 

The reason I am making this a PR and not simply publishing it to the Portal Loop is that it allows for further code changes down the line. `examples/` get loaded into genesis first, and all other replayed addpkg transactions after that, making it possible to have "upgradeable" realms on Portal Loop.

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [x] Added references to related issues and PRs
- [x] Provided any useful hints for running manual tests
- [x] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Comment by leohhhn** [comment]
Screenshot 2024-08-30 at 16 08 03

File: examples/gno.land/r/leon/config/config.gno
Comment (deelawn) [inline examples/gno.land/r/leon/config/config.gno:1]: Is the idea that this should be used whenever someone wants to deploy a home realm?

Comment (leohhhn) [reply examples/gno.land/r/leon/config/config.gno:1]: No, not necessarily. This is just my personal preference, and it was inspired by @moul.

The r/username folder is really up to the person themselves to choose what they want to have inside of it :)

Comment (moul) [reply examples/gno.land/r/leon/config/config.gno:1]: Let's merge only core contributors with a genuine investment in creating high-quality code through pull requests with minimal feedback. We can adjust this approach if necessary.

**Review by moul — APPROVED** [review]

**Review by deelawn — APPROVED** [review]

**Comment by leohhhn** [comment]
Up on https://gno.land/r/leon/home

**Pull request merged**